src/masternode/masternode-utils.cpp

```diff
@@ -27,6 +27,17 @@ void CMasternodeUtils::ProcessMasternodeConnections(CConnman& connman)
     privateSendClient.GetMixingMasternodesInfo(vecDmns);
 #endif // ENABLE_WALLET
 
+    // Don't disconnect masternode connections when we have less then the desired amount of outbound nodes
+    int nonMasternodeCount = 0;
+    connman.ForEachNode(CConnman::AllNodes, [&](CNode* pnode) {
+        if (!pnode->fInbound && !pnode->fFeeler && !pnode->m_manual_connection && !pnode->fMasternode) {
+            nonMasternodeCount++;
+        }
+    });
+    if (nonMasternodeCount < connman.GetMaxOutboundNodeCount()) {
+        return;
+    }
+
     connman.ForEachNode(CConnman::AllNodes, [&](CNode* pnode) {
         if (pnode->fMasternode && !connman.IsMasternodeQuorumNode(pnode)) {
```

Comment:
Wouldn't adding `&& !pnode->m_manual_connection` here be a more direct fix (for tests)?

Reply:
The dropped connections are not manual connections in the tests. In tests, all masternodes are manually connected to the controller node and no other connection is manually added. Later LLMQ connections are formed between MNs, which are then dropped when the large reorg happens in the dip4 coinbase tests. When in at the same time a MN has requested a block from another MN, it won't properly fall back to re-requesting from the controller node (even though it announced the block via a compact blocks).

As said...the bug I tried to fix is actually something different. I then decided to implicitely fix the symptom by improving masternode connection handling (no reason to drop MN connections when we don't have good alternatives).